src/ucp/proto/rndv.c

```diff
@@ -1118,10 +1190,14 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_receive, (worker, rreq, rndv_rts_hdr),
                 goto out;
             }
         }
-        /* put protocol is allowed - register receive buffer memory for rma */
-        ucs_assert(rndv_rts_hdr->size <= rreq->recv.length);
-        ucp_request_recv_buffer_reg(rreq, ep_config->key.rma_bw_md_map,
-                                    rndv_rts_hdr->size);
+
+        if ((rndv_mode == UCP_RNDV_MODE_PUT_ZCOPY) ||
```

Comment:
looks like now we do not register RX buffer for PUT protocol in case of `UCP_RNDV_MODE_PUT_AUTO` + non-CUDA memory

Reply:
Correct. This is PR does not change any behavior for the HOST memory.  Today,  it does GET  by default for HOST memory. it sends RTR w/o registering RX buffer to switch to Active message rndv it fails to do GET (https://github.com/openucx/ucx/blob/master/src/ucp/proto/rndv.c#L473)

Reply:
@brminich are we ok here?

Reply:
ok